src/BlockArea.cpp

```diff
@@ -517,7 +517,7 @@ bool cBlockArea::Write(cForEachChunkProvider * a_ForEachChunkProvider, int a_Min
 		LOGWARNING("%s: MinBlockY less than zero, adjusting to zero", __FUNCTION__);
 		a_MinBlockY = 0;
 	}
-	else if (a_MinBlockY > cChunkDef::Height - m_Size.y)
+	else if (a_MinBlockY >= cChunkDef::Height - m_Size.y)
```

Comment:
I'm not sure about this change either; if it is indeed needed, then the assignment on line 523 is wrong.